presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -140,7 +146,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
                     right,
                     node.getCriteria(),
                     node.getOutputSymbols(),
-                    node.getFilter().map(this::simplifyExpression),
+                    node.getFilter().map(this::simplifyExpression).map(this::simplifyFilterPredicate),
```

Comment:
put your `.map(..)` in separate line. Lets keep a convention of having one `.map` per line.